matcher/namespaces.go

```diff
@@ -64,6 +63,8 @@ type Namespaces interface {
 	Close()
 }
 
+type rulesNamespace rules.Namespace
```

Comment:
I think the forked genny allows you to provide a custom type for a generic type during code generation so you can avoid the typedefs.

Reply:
Yes it does however the rename of certain types using `gorename` requires that the type is accessible in the current package that the symbol is being renamed.  Hence needing this type of typedef for now.